src/main/java/com/uid2/shared/store/CloudPath.java

```diff
@@ -0,0 +1,57 @@
+package com.uid2.shared.store;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+public class CloudPath {
+    private final Path path;
+
+    public CloudPath(String path) {
+        this(Paths.get(path));
+    }
+
+    public CloudPath(Path path) {
+        this.path = path;
+    }
+
+    public CloudPath getParent() {
+        return new CloudPath(path.getParent());
+    }
+
+    public CloudPath resolve(String other) {
+        return new CloudPath(path.resolve(other));
+    }
+
+    public CloudPath resolve(CloudPath other) {
+        return new CloudPath(path.resolve(other.path));
+    }
+
+    public CloudPath getFileName() {
+        return new CloudPath(path.getFileName());
+    }
+
+    @Override
+    public String toString() {
+        String path = this.path.toString();
+
+        // Java's Path always converts everything to path standards of the OS it's running on.
+        // The paths are actually S3 paths so should always be Linux style, so we need to convert them when on Windows
+        return isWindows ? path.replace("\\", "/") : path ;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Include `isWindows` in the check (and update hashCode as well)

Reply:
Why would it be in check. It's a static, not specific to any instance.

Reply:
Oh yes, I missed the static, nvm